app/models/base_profile.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class BaseProfile < ApplicationRecord
+  self.abstract_class = true
+  scope :kept, -> { undiscarded.joins(:user).merge(User.kept) }
```

Comment:
This makes it very difficult to unscope the default scope. As far as I can tell, the only way is `unscoped`, which comes with it's own issues. 
What do you think about adding an `after_discard` to the user which discards all the profiles instead?

Reply:
Done